Sming/SmingCore/Network/Ssl/SslValidator.h

```diff
@@ -16,20 +16,64 @@
 #ifndef _SMING_CORE_SSLVALIDATOR_H_
 #define _SMING_CORE_SSLVALIDATOR_H_
 
-#ifdef ENABLE_SSL
-
 #include "ssl/ssl.h"
 #include "ssl/tls1.h"
 
 #include <functional>
+#include "WVector.h"
+
+#include "SslFingerprints.h"
 
+/** @brief Validator callback function
+ *  @param ssl Contains certificate to validate (may be NULL)
+ *  @param data Data for the callback to use
+ *  @retval bool true if validation succeeded
+ *  @note Callback must ALWAYS release any allocate memory before returning.
+ *  If called with ssl = NULL then just release memory and return false.
+ */
 typedef std::function<bool(SSL* ssl, void* data)> SslValidatorCallback;
 
-bool sslValidateCertificateSha1(SSL* ssl, void* data);
+struct SslValidator {
+	SslValidatorCallback callback;
+	void* data; ///< Callback-specific data, e.g. fingerprint to compare against
+};
+
+class SslValidatorList : private Vector<SslValidator>
+{
+public:
+	~SslValidatorList()
+	{
+		// Make sure memory gets released
+		validate(nullptr);
+	}
+
+	bool add(SslValidatorCallback callback, void* data)
+	{
+		return Vector::add(SslValidator{callback, data});
```

Comment:
Annoyingly, `Vector::add` returns true without actually checking if it succeeded. `WVector.h` needs an overhaul! Do you want it fixed in this PR, or a separate one?

Reply:
> `WVector.h` needs an overhaul! Do you want it fixed in this PR, or a separate one?

A seperate one, please.